src/python/zquantum/core/trackers.py

```diff
@@ -0,0 +1,139 @@
+        self, circuits: Sequence[Circuit], n_samples: List[int]
+    ) -> List[Measurements]:
+        """Run a set of circuits and measure a certain number of bitstrings.
+
+        Args:
+            circuits: The circuits to execute.
+            n_samples: The number of samples to collect for each circuit.
+        """
+        measurements = self.inner_backend.run_circuitset_and_measure(
+            circuits, n_samples
+        )
+        for circuit, measurement in zip(circuits, measurements):
+            self.record_raw_measurement_data(circuit, measurement)
+        self.save_raw_measurement_data()
+        return measurements
+
+    def get_bitstring_distribution(
+        self, circuit: Circuit, n_samples: int
+    ) -> BitstringDistribution:
+        """Calculates a bitstring distribution.
+
+        This function is a wrapper around `get_measurement_outcome_distribution`
+        needed for backward-compatibility.
+
+        Args:
+            circuit: quantum circuit to be executed.
+
+        Returns:
+            Probability distribution of getting specific bistrings.
+        """
+        return self.inner_backend.get_bitstring_distribution(circuit, n_samples)
+
+    def get_measurement_outcome_distribution(
+        self, circuit: Circuit, n_samples: int
+    ) -> MeasurementOutcomeDistribution:
+        """Calculates a measurement outcome distribution.
+
+        Args:
+            circuit: quantum circuit to be executed.
+
+        Returns:
+            Probability distribution of getting specific bistrings.
+
+        """
+        return self.inner_backend.get_measurement_outcome_distribution(
+            circuit, n_samples
+        )
+
+    def record_raw_measurement_data(
+        self, circuit: Circuit, measurement: Measurements
+    ) -> None:
+        """Append data from a measurement to self.raw_measurement_data.
+
+        Args:
+            circuit: Implemented circuit.
+            measurement: Implemented measurement.
+        """
+        raw_data_dict = {
+            "device": self.type,
+            "circuit": to_dict(circuit),
+            "counts": measurement.get_counts(),
+            "number_of_gates": len(circuit.operations),
+            "number_of_shots": len(measurement.bitstrings),
+        }
+        if self.record_bitstrings:
+            raw_data_dict["bitstrings"] = [
+                list(map(int, list(bitstring))) for bitstring in measurement.bitstrings
+            ]
+        self.raw_measurement_data.append(raw_data_dict)
+
+    def save_raw_measurement_data(self) -> None:
+        with open(self.file_name, "w+") as f:
+            data = {
+                "schema": SCHEMA_VERSION + "-raw-measurement-data",
+                "raw-measurement-data": self.raw_measurement_data,
+            }
+            f.write(json.dumps(data))
+        self.raw_measurement_data = []
+
+    def rename_raw_measurement_data_file(self) -> None:
+        rename(self.file_name, "raw-measurement-data.json")
```

Comment:
Do I understand correctly that the motivation for this method was to let the user to store the data under a different file name? WDYT about removing this method and allowing the user to pass a file object to `save_raw_measurement_data`?